src/async-components/views/dialogs/security/CreateKeyBackupDialog.tsx

```diff
@@ -76,23 +76,23 @@ export default class CreateKeyBackupDialog extends React.PureComponent<IProps, I
         });
         const cli = MatrixClientPeg.safeGet();
         try {
-            // We don't want accessSecretStorage to create a backup for us - we
-            // will create one ourselves in the closure we pass in by calling
-            // resetKeyBackup.
-            const setupNewKeyBackup = false;
             const forceReset = false;
-
-            await accessSecretStorage(
-                async (): Promise<void> => {
+            // Check if 4S already setup
+            const secretStorageAlreadySetup = await cli.hasSecretStorageKey();
+
+            await accessSecretStorage(async (): Promise<void> => {
```

Comment:
Yes, I added 2 cases. If the secret storage is not set we call `accessSecretStorage`, if there is one we instead call `withSecretStorageKeyCache`. 

When calling `withSecretStorageKeyCache` we first do a call to `cli.secretStorage.get("m.megolm_backup.v1")`, that will require the user to provide the passphrase/key. If the user is not able to give it then it will fail and the backup will not be created (to avoid desynchronisation of secret storage).
I added a new integration test step to ensure that